src/stores/WalletStore.ts

```diff
@@ -38,24 +36,99 @@ class WalletStore extends PersistableStore {
     return web3Modal.cachedProvider
   }
 
-  async connect(clearCachedProvider = false) {
+  private async getUserNetworkName(provider: Web3Provider) {
+    return (await provider.getNetwork()).name
+  }
+
+  private async isNetworkRight(provider: Web3Provider) {
+    const network = env.VITE_ETH_NETWORK
```

Comment:
Maybe we should create `const preferedNetwork = env.VITE_ETH_NETOWRK` above this `class`, so we don't need to use `env` every time?

Reply:
sure, nice idea. ty